app/components/newsletter-signup/component.js

```diff
@@ -9,6 +19,24 @@ export default Component.extend({
   successMsg: null,
   errorMsg: null,
 
+  init() {
+    this._super(...arguments);
+    let changeset = new Changeset(
+      { legal: true },
+      lookupValidator(validations),
+      validations
+    );
+    set(this, 'changeset', changeset);
+  },
+
+  disableSubmit: computed('changeset.legal', function() {
```

Comment:
if this value is based on errors, you should add `changeset.errors.length` to the list of dependent properties

Reply:
Actually it doesn't ever touch `changeset.errors`, so I've just removed the reference to it on the next line.